tendrl/commons/flows/create_cluster/__init__.py

```diff
@@ -232,3 +232,10 @@ def run(self):
         finally:
             # releasing nodes if any exception came
             create_cluster_utils.release_node_lock(self.parameters)
+
+            # Persist the cluster's extra details
+            NS.tendrl.objects.Cluster(
```

Comment:
This finally block can be reached even if there is an exception, do you really want to save network details in case of exception?